students/ahwall/Lesson06Homework/Lesson06HW_Git/MailRoomPt3.py

```diff
@@ -0,0 +1,167 @@
+#!/usr/bin/env python
+
+'''
+Add ValueError Exception to entries where an float is required
+'''
+import sys
+
+donors = [("Donor A", [10, 20, 30, 40]),
+          ("Donor B", [50, 60, 70, 80]),
+          ("Donor C", [90, 100, 110, 120])]
+
+donordict = {'Donor': ['Andrew Wall', 'Jeff Bezos', 'Bill Gates'],
+             'Donation': [100, 260, 420]}
+
+
+def thank_you():
+    ty_letter = input("Please enter the full name of the donor\n\n"
+                      "\tOr type 'list' to view donor name list:  ")
+    return ty_letter
+    if ty_letter == 'list':  # Works
+        print("\nDonor List")
+        print("----------")
+        list_of_donors = []
+        for items in donors:
+            list_of_donors = ("{: ^10s}".format(items[0]))
+            print(list_of_donors)
+        # loops through to print All Donors in Donor List
+        print()
+        thank_you()
+        return list_of_donors
+    else:
+        i = 0
+        while i < 1:
+            for item in donors:
+                DonorInList = False
+                if item[0].count(ty_letter) == 1:
+                    DonorInList = True  # works
+                    break
+            if DonorInList is True:
+                j = 0
```

Comment:
I don't know if I already commented on this, but man this is complicated. If you had time to refactor it'd be great to break this up and simplify. In Python for loops pretty much can always take the place of using i = 0 and i += 1.